adapters/thetradedesk/thetradedesk.go

```diff
@@ -0,0 +1,177 @@
+package thetradedesk
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+)
+
+const PREBID_INTEGRATION_TYPE = "1"
+
+type adapter struct {
+	bidderEndpoint string
+}
+
+type ExtImpBidderTheTradeDesk struct {
+	adapters.ExtImpBidder
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	pubID, err := getPublisherId(request.Imp)
+
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	for i := 0; i < len(request.Imp); i++ {
+		imp := request.Imp[i]
+
+		if imp.Banner != nil {
+			if len(imp.Banner.Format) > 0 {
+				firstFormat := imp.Banner.Format[0]
+				imp.Banner.H = &firstFormat.H
+				imp.Banner.W = &firstFormat.W
+
+			}
+		}
+	}
```

Comment:
You're going to need to set `request.Imp` to a new slice of modified imps since you're modifying copies of each imp. This would become apparent if you had a JSON test that contains multiple formats in `imp.banner.format` where the first format value is different from what is initially in `imp.banner.h` and `imp.banner.w` so that the first format overwrites `imp.banner.h` and `imp.banner.w`. Such a test would fail at the moment so please add that first.

Then to set the `request.imp` you can weave these statements into this loop:
```
modifiedImps := make([]openrtb2.Imp, 0, len(request.Imp))
modifiedImps = append(modifiedImps, imp)
request.Imp = modifiedImps
```

Reply:
test case added.